diff --git a/extern/test_bmi_c/src/bmi_test_bmi_c.c b/extern/test_bmi_c/src/bmi_test_bmi_c.c
index 71e11c62e..47bf4e6a2 100644
--- a/extern/test_bmi_c/src/bmi_test_bmi_c.c
+++ b/extern/test_bmi_c/src/bmi_test_bmi_c.c
@@ -729,7 +729,11 @@ int read_file_line_counts(const char* file_name, int* line_count, int* max_line_
         return -1;
     }
     int seen_non_whitespace = 0;
-    char c;
+    int c; //EOF is a negative constant...and char may be either signed OR unsigned
+         //depending on the compiler, system, achitectured, ect.  So there are cases
+         //where this loop could go infinite comparing EOF to unsigned char
+         //the return of fgetc is int, and should be stored as such!
+         //https://stackoverflow.com/questions/35356322/difference-between-int-and-char-in-getchar-fgetc-and-putchar-fputc
     for (c = fgetc(fp); c != EOF; c = fgetc(fp)) {
         // keep track if this line has seen any char other than space or tab
         if (c != ' ' && c != '\t' && c != '\n')
diff --git a/extern/test_bmi_cpp/src/test_bmi_cpp.cpp b/extern/test_bmi_cpp/src/test_bmi_cpp.cpp
index de19b9e86..c371879e4 100644
--- a/extern/test_bmi_cpp/src/test_bmi_cpp.cpp
+++ b/extern/test_bmi_cpp/src/test_bmi_cpp.cpp
@@ -426,7 +426,11 @@ void TestBmiCpp::read_file_line_counts(std::string file_name, int* line_count, i
     throw std::runtime_error("Configuration file does not exist." SOURCE_LOC);
   }
   int seen_non_whitespace = 0;
-  char c;
+  int c; //EOF is a negative constant...and char may be either signed OR unsigned
+         //depending on the compiler, system, achitectured, ect.  So there are cases
+         //where this loop could go infinite comparing EOF to unsigned char
+         //the return of fgetc is int, and should be stored as such!
+         //https://stackoverflow.com/questions/35356322/difference-between-int-and-char-in-getchar-fgetc-and-putchar-fputc
   for (c = fgetc(fp); c != EOF; c = fgetc(fp)) {
     // keep track if this line has seen any char other than space or tab
     if (c != ' ' && c != '\t' && c != '\n')
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 78b7e0920..722da840d 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -285,6 +285,7 @@ if(NGEN_ACTIVATE_ROUTING)
             test_routing_pybind
             1
             routing/Routing_Py_Bind_Test.cpp
+            NGen::core # for filechecker utility
             NGen::routing
             pybind11::embed
     )
diff --git a/test/core/nexus/NexusRemoteTests.cpp b/test/core/nexus/NexusRemoteTests.cpp
index 524b7da05..2e9e48693 100644
--- a/test/core/nexus/NexusRemoteTests.cpp
+++ b/test/core/nexus/NexusRemoteTests.cpp
@@ -68,6 +68,11 @@ void Nexus_Remote_Test::TearDown()
 //to a downstream remote nexus.
 TEST_F(Nexus_Remote_Test, TestInit0)
 {
+    if ( mpi_num_procs < 2 )
+    {
+    	GTEST_SKIP();
+    }
+
     HY_PointHydroNexusRemote::catcment_location_map_t loc_map;
 
     std::shared_ptr<HY_PointHydroNexusRemote> nexus;
@@ -409,6 +414,11 @@ TEST_F(Nexus_Remote_Test, Test4R2S2LS)
 
 TEST_F(Nexus_Remote_Test, TestDeadlock1)
 {
+    if ( mpi_num_procs < 2 )
+    {
+    	GTEST_SKIP();
+    }
+
     HY_PointHydroNexusRemote::catcment_location_map_t loc_map;
 
     std::shared_ptr<HY_PointHydroNexusRemote> nexus1;
diff --git a/test/data/routing/ngen_routing_config_unit_test.yaml b/test/data/routing/ngen_routing_config_unit_test.yaml
index 80f4a40ab..a93eaf5c0 100644
--- a/test/data/routing/ngen_routing_config_unit_test.yaml
+++ b/test/data/routing/ngen_routing_config_unit_test.yaml
@@ -9,7 +9,7 @@ network_topology_parameters:
     supernetwork_parameters:
         #----------
         geo_file_type: HYFeaturesNetwork 
-        geo_file_path: ./test/data/routing/gauge_01073000.gpkg 
+        geo_file_path: ../../test/data/routing/gauge_01073000.gpkg
         mask_file_path: # domain/unit_test_noRS/coastal_subset.txt
         synthetic_wb_segments:
             #- 4800002
@@ -21,8 +21,8 @@ network_topology_parameters:
         break_network_at_waterbodies: False 
         level_pool:
             #----------
-            level_pool_waterbody_parameter_file_path: ./test/data/routing/gauge_01073000.gpkg 
-            reservoir_parameter_file                : ./test/data/routing/gauge_01073000.gpkg
+            level_pool_waterbody_parameter_file_path: ../../test/data/routing/gauge_01073000.gpkg
+            reservoir_parameter_file                : ../../test/data/routing/gauge_01073000.gpkg
         #rfc:
             #----------
             #reservoir_parameter_file                : domain/reservoir_index_AnA.nc
@@ -58,11 +58,11 @@ compute_parameters:
         #----------
         qts_subdivisions            : 12
         dt                          : 300 # [sec]
-        qlat_input_folder           : ./test/data/routing/ 
+        qlat_input_folder           : ../../test/data/routing/
         qlat_file_pattern_filter    : "nex-*"
-        nexus_input_folder          : ./test/data/routing/ 
+        nexus_input_folder          : ../../test/data/routing/
         nexus_file_pattern_filter   : "nex-*" #OR "*NEXOUT.parquet" OR "nex-*"
-        binary_nexus_file_folder    : ./test/data/routing/ # this is required if nexus_file_pattern_filter="nex-*"
+        binary_nexus_file_folder    : ../../test/data/routing/ # this is required if nexus_file_pattern_filter="nex-*"
         #coastal_boundary_input_file : channel_forcing/schout_1.nc  
         nts                         : 8640 #288 for 1day
         max_loop_size               : 720 # [hr]  
@@ -98,6 +98,6 @@ output_parameters:
         #wrf_hydro_channel_output_source_folder: ./
     chanobs_output:
         #----------
-        chanobs_output_directory: ./test/data/routing/ 
+        chanobs_output_directory: ../../test/data/routing/
         chanobs_filepath        : Chanobs.nc
-    lakeout_output: ./test/data/routing/
+    lakeout_output: ../../test/data/routing/
diff --git a/test/realizations/Formulation_Manager_Test.cpp b/test/realizations/Formulation_Manager_Test.cpp
index 7fff4f9fb..230125faa 100644
--- a/test/realizations/Formulation_Manager_Test.cpp
+++ b/test/realizations/Formulation_Manager_Test.cpp
@@ -69,11 +69,12 @@ class Formulation_Manager_Test : public ::testing::Test {
 
     std::vector<std::string> path_options = {
             "",
+            "../",
+            "../../",
             "./test/",
             "../test/",
-            "../../test/",
-            "../",
-            "../../"
+            "../../test/"
+
     };
 
     std::string fix_paths(std::string json)
@@ -84,6 +85,17 @@ class Formulation_Manager_Test : public ::testing::Test {
                 "./data/forcing/cat-27_2015-12-01 00_00_00_2015-12-30 23_00_00.csv",
                 "./data/forcing/cat-27115-nwm-aorc-variant-derived-format.csv"
         };
+        std::vector<std::string> v = {};
+        for(unsigned int i = 0; i < path_options.size(); i++){
+                v.push_back( path_options[i] + "data/forcing" );
+        }
+        std::string dir = utils::FileChecker::find_first_readable(v);
+        if(dir != ""){
+                std::string remove = "\"./data/forcing/\"";
+                std::string replace = "\""+dir+"\"";
+                //std::cerr<<"TRYING TO REPLACE DIRECTORY... "<<remove<<" -> "<<replace<<std::endl;
+                boost::replace_all(json, remove , replace);
+        }
         for (unsigned int i = 0; i < forcing_paths.size(); i++) {
           if(json.find(forcing_paths[i]) == std::string::npos){
             continue;
@@ -95,7 +107,7 @@ class Formulation_Manager_Test : public ::testing::Test {
           std::string right_path = utils::FileChecker::find_first_readable(v);
           if(right_path != forcing_paths[i]){
             std::cerr<<"Replacing "<<forcing_paths[i]<<" with "<<right_path<<std::endl;
-            json = json.replace(json.find(forcing_paths[i]), forcing_paths[i].length(), right_path);
+            boost::replace_all(json, forcing_paths[i] , right_path);
           }
         }
         return json;
diff --git a/test/routing/Routing_Py_Bind_Test.cpp b/test/routing/Routing_Py_Bind_Test.cpp
index 2831368e5..a89da36c9 100644
--- a/test/routing/Routing_Py_Bind_Test.cpp
+++ b/test/routing/Routing_Py_Bind_Test.cpp
@@ -1,6 +1,7 @@
 #ifdef ROUTING_PYBIND_TESTS_ACTIVE
 #include "gtest/gtest.h"
 #include "Routing_Py_Adapter.hpp"
+#include "FileChecker.h"
 #include <string>
 #include <pybind11/embed.h>
 #include <pybind11/stl.h>
@@ -33,8 +34,12 @@ TEST_F(RoutingPyBindTest, TestRoutingPyBind)
   //std::vector<double> nexus_values_vec{1.1, 2.2, 3.3, 4.4, 5.5};
 
   //SET THESE AS INPUTS
-  std::string t_route_config_file_with_path = "./test/data/routing/ngen_routing_config_unit_test.yaml";
-
+  std::vector<std::string> paths = {
+          "./test/data/routing/ngen_routing_config_unit_test.yaml",
+          "../test/data/routing/ngen_routing_config_unit_test.yaml",
+          "../../test/data/routing/ngen_routing_config_unit_test.yaml"
+  };
+  std::string t_route_config_file_with_path = utils::FileChecker::find_first_readable(paths);
   //Note: Currently, delta_time is set in the t-route yaml configuration file, and the
   //number_of_timesteps is determined from the total number of nexus outputs in t-rout
   //It is recommended to still pass these values to the routing_py_adapter object in
